src/components/inputs/localized-money-input/localized-money-input.js

```diff
@@ -67,6 +67,15 @@ class LocalizedInput extends React.Component {
     //
     // eslint-disable-next-line no-param-reassign
     event.target.currency = this.props.currency;
+
+    // We manipulate the event to remove any characters after the currencyCode.
+    // This is for use by Formik as it expects the input name to be of the form:
+    // "price.EUR"
+    // But the onChange from the MoneyInput produces an event with name of the form:
+    // "price.EUR.amount"
+    //
+    // eslint-disable-next-line no-param-reassign
+    event.target.name = event.target.name.replace(/(.*)(\..*)$/, '$1');
```

Comment:
We should probably change the element name instead as we also need these event names on the blur event.

You should be able to change the implementation of `LocalizedMoneyInput.getName`.

Reply:
`LocalizedMoneyInput.getName` works correctly:

```
export const getName = (namePrefix, language) =>
  namePrefix ? `${namePrefix}.${language}` : undefined;
```

The `.amount` and `.currencyCode` suffices come from the `MoneyInput`.

Reply:
I prefer this too. It's simpler than changing the event names.